### PR TITLE
adding EKS Account ID for security measures when using  for AMIs

### LINKED ISSUE #732: Contain AMI Search Based on EKS Account ID for AL2 images
**Why do you want this feature?**
This could lead to EKS clusters being created with non AWS owned AMIs.

**What feature/behavior/change do you want?**
Current search could include AMIs not AWS owned but made public using the same tags with a more recent created at, the request is to add an additional check for Image searching using `auto` and the `make generate-ami` list to only be AWS owned. *Ubuntu images can be from any account for now*

### PR DESCRIPTION
**Pull request by christopherhein**
Signed-off-by: Christopher Hein <me@chrishein.com>

### Description
closes #732 

This checks the right Account ID for AL2 AMIs so that they always come from the EKS team.

### Checklist

- [x] Code compiles correctly (i.e `make build`)
- [x] All unit tests passing (i.e. `make test`)
- [x] All integration tests passing (i.e. `make integration-test`)

### CODE REVIEW

File: pkg/ami/api.go [hunk 1]
```diff
@@ -86,6 +86,10 @@ func FindImage(api ec2iface.EC2API, namePattern string) (string, error) {
 		},
 	}
 
+	if imageFamily == "ImageFamilyAmazonLinux2" {
```
Comment (errordeveloper) [inline pkg/ami/api.go]: I think we should match family to owner elsewhere, perhaps we can look it up from another map. But from the standpoint if this function, it's unaware of families of what now, so we just want to pass the owner to it.

**Comment by errordeveloper** [comment]
It would be also great to not spoil the coverage, this package has pretty good coverage and we should keep it that way. So please take a look if you can add a meaningful test for this!

File: pkg/ami/api.go (hunk 1 above)
Comment (errordeveloper) [reply pkg/ami/api.go]: Both owner IDs are documented here already:

https://github.com/weaveworks/eksctl/blob/93cc5eb09b06a7b60ae6cba123fafd82c001be71/pkg/ami/api.go#L17-L21

Please use the Ubuntu one also, and after you have a lookup map, remove those comments.

**Comment by christopherhein** [comment]
Once tests pass let me know what you think @errordeveloper

**Comment by christopherhein** [comment]
@errordeveloper everything looks good now on this if you get a second to merge.

**Comment by martina-if** [comment]
LGTM

**Comment by nrdlngr** [comment]
Any update on this?

File: pkg/ami/api.go [hunk 2]
```diff
@@ -60,8 +66,9 @@ func IsAvailable(api ec2iface.EC2API, id string) (bool, error) {
 // FindImage will get the AMI to use for the EKS nodes by querying AWS EC2 API.
 // It will only look for images with a status of available and it will pick the
 // image with the newest creation date.
-func FindImage(api ec2iface.EC2API, namePattern string) (string, error) {
+func FindImage(api ec2iface.EC2API, namePattern string, imageFamily string) (string, error) {
 	input := &ec2.DescribeImagesInput{
+		Owners: []*string{aws.String(ImageFamilyToAccountID[imageFamily])},
```
Comment (errordeveloper) [inline pkg/ami/api.go]: This a little too optimistic, we should check if the key exists, so we don't panic if it doesn't.

File: pkg/ami/api.go [hunk 3]
```diff
@@ -60,8 +66,9 @@ func IsAvailable(api ec2iface.EC2API, id string) (bool, error) {
 // FindImage will get the AMI to use for the EKS nodes by querying AWS EC2 API.
 // It will only look for images with a status of available and it will pick the
 // image with the newest creation date.
-func FindImage(api ec2iface.EC2API, namePattern string) (string, error) {
+func FindImage(api ec2iface.EC2API, namePattern string, imageFamily string) (string, error) {
```
Comment (errordeveloper) [inline pkg/ami/api.go]: I think map lookup should be all in the same place, let's just make it like this here:
```suggestion
func FindImage(api ec2iface.EC2API, namePattern, accountID string) (string, error) {
```

File: pkg/ami/api.go (hunk 2 above)
Comment (errordeveloper) [inline pkg/ami/api.go]:
```suggestion
		Owners: []*string{&accountID},
```

Comment (errordeveloper) [reply pkg/ami/api.go]: Please move the look-up  into the caller, and the map declaration should live in the same file also:

https://github.com/weaveworks/eksctl/blob/3d20b573fcc5b495ba6729ba4affdc25a285639a/pkg/ami/auto_resolver.go#L50

**Comment by errordeveloper** [comment]
We should be able to include this in the next release, but due to undergoing API changes, we cannot cut it before all of [milestone issues](https://github.com/weaveworks/eksctl/milestone/11) are closed, as we are half-way-through.

**Comment by errordeveloper** [comment]
@christopherhein I'll have to finish this up myself, as we are looking to cut 0.1.30 today.

File: pkg/ami/auto_resolver.go [hunk 4]
```diff
@@ -50,17 +56,23 @@ type AutoResolver struct {
...
 		if !ok {
 			logger.Critical("image family %s doesn't support GPU image class", imageFamily)
 			return "", NewErrFailedResolution(region, version, instanceType, imageFamily)
 		}
 	}
 
-	id, err := FindImage(r.api, p, imageFamily)
+	ownerAccount, knownOwner := ImageFamilyToAccountID[imageFamily]
+	if !knownOwner {
+		logger.Critical("unable to determin account owner for image family %s", imageFamily)
```
Comment (martina-if) [inline pkg/ami/auto_resolver.go]: typo: ~determin~ determine

Comment (martina-if) [reply pkg/ami/auto_resolver.go]: Also: "determine the account owner..." might read better

Comment (errordeveloper) [reply pkg/ami/auto_resolver.go]: 👍

**Pull request merged**